.github/workflows/pip-build-osx.yml

```diff
@@ -13,7 +13,18 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: ['3.11']
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        include:
+          - python-version: '3.7'
+            numpy-version: '1.15.0'
+          - python-version: '3.8'
+            numpy-version: '1.18.0'
+          - python-version: '3.9'
+            numpy-version: '1.20.0'
+          - python-version: '3.10'
+            numpy-version: '1.22.0'
+          - python-version: '3.11'
+            numpy-version: '1.24.0'
```

Comment:
The tensorflow package for python 3.11 requires numpy (<1.24,>=1.22). If gudhi wants >=1.24, won't that cause problems?